### PR TITLE
docs: add docs for torch profiler support [MLG-179]

### PR DESCRIPTION
**Pull request by azhou-determined**
## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: docs/training/apis-howto/api-pytorch-ug.rst [hunk 1]
```diff
@@ -505,6 +505,40 @@ you find that the built-in context.DataLoader() does not support your use case.
 
 See the :mod:`determined.pytorch.samplers` for details.
 
+Profiling
+---------
+
+Determined provides integrated support for the native PyTorch profiler, `torch-tb-profiler
```
Comment (tara-det-ai) [inline docs/training/apis-howto/api-pytorch-ug.rst:511]: Determined provides support for the native PyTorch profiler....

(is "integrated" assumed since having integrated it we are showing them how to use it?)

File: docs/training/apis-howto/api-pytorch-ug.rst [hunk 2]
```diff
@@ -505,6 +505,40 @@ you find that the built-in context.DataLoader() does not support your use case.
...
+<https://github.com/pytorch/kineto/tree/main/tb_plugin>`_. This can be configured with
```
Comment (tara-det-ai) [inline docs/training/apis-howto/api-pytorch-ug.rst:512]: You can configure this by calling...

(rather than "This can be configured with...)

File: docs/training/apis-howto/api-pytorch-ug.rst [hunk 3]
```diff
@@ -505,6 +505,40 @@ you find that the built-in context.DataLoader() does not support your use case.
...
+:meth:`~determined.pytorch.PyTorchTrialContext.set_profiler` from within your Trial's ``__init__``.
+``set_profiler`` accepts the same arguments as the PyTorch plugin's ``torch.profiler.profile``
+method, but Determined will set ``on_trace_ready`` to the appropriate Tensorboard path. Stepping of
```
Comment (tara-det-ai) [inline docs/training/apis-howto/api-pytorch-ug.rst:515]: split it into 2 sentences for readability and apply active voice:

``set_profiler`` accepts the same arguments as the PyTorch plugin's ``torch.profiler.profile`` method. However, Determined sets ``on_trace_ready`` to the appropriate Tensorboard path, and the stepping of the profiler during training is automatically handled.

File: docs/training/apis-howto/api-pytorch-ug.rst [hunk 4]
```diff
@@ -505,6 +505,40 @@ you find that the built-in context.DataLoader() does not support your use case.
...
+the profiler during training will be automatically handled.
+
+This example will profile CPU and GPU activities on batches 3 and 4 (skipping batch 1, warming up on
+batch 2), and repeat for 2 cycles:
```
Comment (tara-det-ai) [inline docs/training/apis-howto/api-pytorch-ug.rst:519]: The following example profiles CPU and GPU activities on batches 3 and 4 (skipping batch 1, warming up on batch 2), and repeats for 2 cycles:
